pandas/core/dtypes/cast.py

```diff
@@ -1162,3 +1162,23 @@ def construct_1d_arraylike_from_scalar(value, length, dtype):
         subarr.fill(value)
 
     return subarr
+
+
+def construct_1d_array_from_listlike(values, dtype='object'):
+    """
+    Transform any list-like object in a 1-dimensional numpy array.
+
+    Parameters
+    ----------
+    values : any iterable which has a len()
+    dtype : dtype, default 'object'
+
+    Returns
+    -------
+    1-dimensional numpy array of dtype "dtype"
```

Comment:
I would rename this to 
``construct_1d_object_array_from_listlike`` and drop the dtype (always make it ``object``).

you can also just add another function for this as well (e.g. have 1 that accepts the dtype as a required parameter ). I also don't think we are actually using dtype anywhere?

Reply:
The array we return is of dtype ``dtype``. So this will raise an error if e.g. ``dtype=int`` and the input contains lists, but this is good because it allows cleaner code (and less checks) in those methods where the actual dtype is unknown.

Reply:
we are not using this, so its very confusing. I would just rather return an ``object`` dtyped array here (or have 2 methods).

Reply:
> we are not using this, so its very confusing

Non sequitur.

> have 2 methods

... with almost perfect duplication of code, one of which perfectly generalizes the other, and without any performance gain?! I thought you liked to _reduce_ code :-)

This method builds a 1d array of type ``dtype`` with given data. If it is impossible, it raises an error. This is pretty straightforward, and whenever ``np.array`` will have an [``ndmax`` parameter](https://github.com/numpy/numpy/issues/10218#issuecomment-351789853), all calls to this method will be replaced with a simple call to ``np.array(..., ndmax=1)``.

Reply:
I guess I will explain again.

We are not using the dtype argument. So just eliminate it. I don't see utility in having it.